src/components/Settings/DatasourceSettings/index.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import ListItem from 'react-md/lib/Lists/ListItem';
+import List from 'react-md/lib/Lists/List';
+import plugins from '../../../data-sources/plugins';
+
+interface IDatasourceSettingsProps {
+  settings: IDataSourceContainer;
+}
+
+interface IDatasourceSettingsState { 
+  selectedIndex: number;
+}
+
+export default class DatasourceSettings extends React.Component<IDatasourceSettingsProps, IDatasourceSettingsState> {
+
+  static uniqueId = 0;
+
+  state = {
+    selectedIndex: 0
+  };
+    
+  constructor(props: IDatasourceSettingsProps) {
+    super(props);
+
+    this.onMenuClick = this.onMenuClick.bind(this);
+    this.getSettings = this.getSettings.bind(this);
+    this.renderSettingsEditor = this.renderSettingsEditor.bind(this);
+    this.renderMenuItems = this.renderMenuItems.bind(this);
+    this.getSettingsEditor = this.getSettingsEditor.bind(this);
+  }
+  
+  getSettingsEditor(element: DataSource): JSX.Element {
```

Comment:
Since this is a complex class, I would add a few comments like /** ... */ to explain the purpose of the methods.
HAE (Here and everywhere) in this class